tests/functional/TestSmokeTest/tests/block_volume_test.go

```diff
@@ -0,0 +1,60 @@
+// +build functional
+
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+package functional
+
+import (
+	"testing"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+	"github.com/heketi/tests"
+)
+
+func TestBlockVolumeOperation(t *testing.T) {
+
+	// Setup the VM storage topology
+	defer teardownCluster(t)
+	setupCluster(t, 3, 4)
+
+	req := &api.BlockVolumeCreateRequest{}
+	//check is it possible to create block volume if  size is greated then block hosting volume
+	req.Size = 101
+
+	_, err := heketi.BlockVolumeCreate(req)
+	tests.Assert(t, err != nil, "expected err != nil, got:", err)
+
+	//check is it possible to create block volume as same size of block hosting volume
+	req.Size = 100
+	_, err = heketi.BlockVolumeCreate(req)
+	tests.Assert(t, err != nil, "expected err != nil, got:", err)
+
+	// 1% is reserved in blockhosting volume, we should be able to create
+	//block volumes of total size 98 GB
+	for i := 1; i <= 49; i++ {
+		req.Size = 2
```

Comment:
Why are you setting req.Size inside the loop?

Reply:
this can be moved outside

Reply:
moved outside of for loop